agent/src/test/java/com/code_intelligence/jazzer/runtime/TraceCmpHooksTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer.runtime;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import org.junit.Test;
+
+public class TraceCmpHooksTest {
+  private static final ExecutorService ES = Executors.newFixedThreadPool(5);
+
+  @Test
+  public void cmpHookShouldHandleConcurrentModifications() throws InterruptedException {
+    String arg = "test";
+    Map<String, Object> map = new LinkedHashMap<>();
+    map.put(arg, arg);
+
+    // Add elements to map asynchronously
+    Function<Integer, Runnable> put = (final Integer num) -> () -> {
+      map.put(String.valueOf(num), num);
+    };
+    for (int i = 0; i < 1_000_000; i++) {
+      ES.submit(put.apply(i));
+    }
+
+    // Call hook
+    try {
+      for (int i = 0; i < 1_000; i++) {
+        TraceCmpHooks.mapGet(null, map, new Object[] {arg}, 1, null);
+      }
+    } catch (UnsatisfiedLinkError ignored) {
+      // Expected error due to missing native library,
```

Comment:
This should be fixable with `MockDriver`. If it isn't, I would like to look into why that's the case.